parser/prism/Translator.cc

```diff
@@ -1638,4 +1598,47 @@ unique_ptr<parser::Regexp> Translator::translateRegexp(pm_string_t unescaped, co
     return make_unique<parser::Regexp>(location, move(parts), move(options));
 }
 
+// Creates a `parser::Mlhs` for either a `PM_MULTI_WRITE_NODE` or `PM_MULTI_TARGET_NODE`.
+template <typename PrismNode> std::unique_ptr<parser::Mlhs> Translator::translateMultiTargetLhs(PrismNode *node) {
+    static_assert(
+        is_same_v<PrismNode, pm_multi_target_node> || is_same_v<PrismNode, pm_multi_write_node>,
+        "Translator::translateMultiTarget can only be used for PM_MULTI_TARGET_NODE and PM_MULTI_WRITE_NODE.");
+
+    auto location = translateLoc(node->base.location);
+
+    // Left-hand side of the assignment
+    auto prismLefts = absl::MakeSpan(node->lefts.nodes, node->lefts.size);
+    auto prismRights = absl::MakeSpan(node->rights.nodes, node->rights.size);
+    auto prismSplat = node->rest;
+
+    NodeVec sorbetLhs{};
+    sorbetLhs.reserve(prismLefts.size() + prismRights.size() + (prismSplat != nullptr ? 1 : 0));
+
+    translateMultiInto(sorbetLhs, prismLefts);
+
+    if (prismSplat != nullptr) {
+        switch (PM_NODE_TYPE(prismSplat)) {
```

Comment:
I wish I could do without this nullcheck, and just be able to handle in another `case`. Opened https://github.com/ruby/prism/issues/3189 to discuss